GMX-Bots/Large-Open-and-Increase-Positions/src/agent.ts

```diff
@@ -0,0 +1,70 @@
+import { Finding, getEthersProvider, HandleTransaction, LogDescription, TransactionEvent } from "forta-agent";
+import { providers } from "ethers";
+import BigNumber from "bignumber.js";
+import { createFinding } from "./utils";
+import NetworkData, { NETWORK_MAP } from "./network";
+import { INCREASE_POSITION_EVENT, UPDATE_POSITION_EVENT, PRICE_PRECISION } from "./constants";
+import NetworkManager from "./network";
+
+const networkManager = new NetworkManager(NETWORK_MAP);
+
+export const initialize = (provider: providers.Provider) => {
+  return async () => {
+    const { chainId } = await provider.getNetwork();
+    networkManager.setNetwork(chainId);
+  };
+};
+
+export const provideTransactionHandler = (
+  networkManager: NetworkData,
+  updatePositionEvent: string,
+  increasePositionEvent: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    // filter transaction logs for increase and update position events
+    const eventLogs: LogDescription[] = txEvent.filterLog(
+      [increasePositionEvent, updatePositionEvent],
+      networkManager.vaultAddress
+    );
+
+    // extract only increaseEventLogs
+    const increaseEventLogs: LogDescription[] = eventLogs.filter((eventLogs) => eventLogs.name == "IncreasePosition");
+
+    // extract only updateEventLogs
+    const updateEventLogs: LogDescription[] = eventLogs.filter((eventLogs) => eventLogs.name == "UpdatePosition");
+
+    increaseEventLogs.forEach((increaseEventLog, index) => {
```

Comment:
The bot is still wrong I mentioned the solution in my previous review but here I am going to explain it again.

You can't split this into `increaseEventLogs` & `updateEventLogs` because by doing that you lose the other beside it seems the opposite to you. The thing is that `UpdatePosition` can occur many times more than `IncreasePosition` so the sizes of this lists are not going to match. 

But you know that after each `IncreasePosition` there is going to be a `UpdatePosition` so you can use the next event in the logs and be sure that it is the corresponding update event each time you find an increase one